app/services/files/create_service.rb

```diff
@@ -9,9 +9,14 @@ def commit
     def validate
       super
 
-      file_name = File.basename(@file_path)
+      if @file_path =~ Gitlab::Regex.directory_traversal_regex
```

Comment:
It already prevents escaping the repo, but it does cause an HTTP 500 instead of gracefully handling the exception. This prevents that.